### DISCUSSION TITLE
Shell scripting - Dispatch

### DISCUSSION THREAD

**Discussion by pjeereddy** [learndevopsonline/learndevopsonline#1277] [Q&A]
I am trying to run dispatch service, but I am getting below error at start service:
![image](https://github.com/learndevopsonline/learndevopsonline/assets/130114098/d2b57e18-9774-4e34-b418-9f82737969d8)
I have checked logs and found below error: 
![image](https://github.com/learndevopsonline/learndevopsonline/assets/130114098/6c3dc761-fb04-4166-b358-ce20ed792ada)
Here is my related code:

**### dispatch.sh:** 
component=dispatch
source common.sh
func_go

**### dispatch.service:**
[Unit]
Description = Dispatch Service
[Service]
User=roboshop
Environment=AMQP_HOST=rabbitmq.jdevops74.online
Environment=AMQP_USER=roboshop
Environment=AMQP_PASS=roboshop123
ExecStart=/app/dispatch
SyslogIdentifier=dispatch

[Install]
WantedBy=multi-user.target

### **_code in common.sh:_**
func_go(){
  log=/tmp/roboshop.log
  echo -e "\e[32m <<<<<<<<<< Install golang >>>>>>>>>>>\e[0m"
  yum install golang -y &>>${log}
  func_exit
  func_prerequisite
  func_exit
  echo -e "\e[32m <<<<<<<<<< download dependencies >>>>>>>>>>>\e[0m"
  go mod init ${component} &>>${log}
  func_exit
  echo -e "\e[32m <<<<<<<<<< get files >>>>>>>>>>>\e[0m"
  go get &>>${log}
  func_exit
  echo -e "\e[32m <<<<<<<<<< build files >>>>>>>>>>>\e[0m"
  go build &>>${log}
  func_exit
  echo -e "\e[32m <<<<<<<<<<< start service >>>>>>>>>>>>> \e[0m"
 func_systemd
 func_exit
}

**function systemd:**
func_systemd(){

  systemctl daemon-reload &>>${log}

  systemctl enable ${component} &>>${log}

  systemctl restart ${component} &>>${log}

}

**function exit:**
func_exit(){
  if [ "$?" -eq "0" ]; then
  echo -e "\e[32m <<<<<<<<<<< success >>>>>>>>>> \e[0m"
 else
   echo -e "\e[31m <<<<<<<<<<< failure >>>>>>>>>> \e[0m"
   fi
}

**function prerequisite:**
func_prerequisite(){
 echo -e "\e[32m>>>>>>>>>>>>>>>> add user>>>>>>>>>>>>>>>> \e[0m"
   id roboshop &>>${log}
   if [ $? -ne 0 ]; then
   useradd roboshop &>>${log}
   fi
   func_exit
   echo -e "\e[32m>>>>>>>>>>>>>>>> cleanup old directory >>>>>>>>>>>>>>>> \e[0m"
   func_exit
   rm -rf /app &>>${log}
   func_exit
   echo -e "\e[32m>>>>>>>>>>>>>>>> create new directory >>>>>>>>>>>>>>>> \e[0m"
   mkdir /app &>>${log}
   func_exit
   echo -e "\e[32m>>>>>>>>>>>>>>>> download application content >>>>>>>>>>>>>>>> \e[0m"
   curl -L -o /tmp/${component}.zip https://roboshop-artifacts.s3.amazonaws.com/${component}.zip &>>${log}
   func_exit
   cd /app &>>${log}
   func_exit
   echo -e "\e[32m>>>>>>>>>>>>>>>> unzip application content >>>>>>>>>>>>>>>> \e[0m"
   unzip /tmp/${component}.zip &>>${log}
   func_exit

   cd /app &>>${log}
}

**Comment by Naresh-vk-243** [comment]
@pjeereddy you haven't wrote the copy command to copy the dispatch.service file, please try adding the

 **`cp dispatch.service /etc/systemd/system/dispatch.service`** to your code
